src/utils/client.js

```diff
@@ -177,7 +182,17 @@ export default class Client {
     steps.forEach((step) => {
       step.servar_fields.forEach((field) => {
         const val = getDefaultFieldValue(field);
-        values[field.servar.key] = field.servar.repeated ? [val] : val;
+        // This can't be done when the canvas is hydrated with initial value
+        // because that runs after the Form onSubmit callback is initialized,
+        // which expects a File, not the base64 supplied by the user
+        const { key, repeated, type } = field.servar;
+        if (type === 'signature' && additionalValues[key]) {
```

Comment:
this check has an edge case if the user tries to set the signature value for a field not present in this form. what if we maintain an invariant that all base 64 strings need to be converted to files instead? and we can typecheck on that directly.

this current logic can also break if the user tries to set a malformed value to a signature field and we attempt to convert that